jdaviz/configs/imviz/plugins/coords_info/coords_info.py

```diff
@@ -12,23 +24,47 @@
 class CoordsInfo(TemplateMixin):
     template_file = __file__, "coords_info.vue"
     icon = Unicode("").tag(sync=True)
-    pixel_prefix = Unicode("Pixel").tag(sync=True)
-    pixel = Unicode("").tag(sync=True)
-    value = Unicode("").tag(sync=True)
-    world_label_prefix = Unicode("\u00A0").tag(sync=True)
-    world_label_prefix_2 = Unicode("\u00A0").tag(sync=True)
-    world_label_icrs = Unicode("\u00A0").tag(sync=True)
-    world_label_deg = Unicode("\u00A0").tag(sync=True)
-    world_ra = Unicode("").tag(sync=True)
-    world_dec = Unicode("").tag(sync=True)
-    world_ra_deg = Unicode("").tag(sync=True)
-    world_dec_deg = Unicode("").tag(sync=True)
-    unreliable_world = Bool(False).tag(sync=True)
-    unreliable_pixel = Bool(False).tag(sync=True)
+
+    row1a_title = Unicode("").tag(sync=True)
+    row1a_text = Unicode("").tag(sync=True)
+    row1b_title = Unicode("").tag(sync=True)
+    row1b_text = Unicode("").tag(sync=True)
+    row1_unreliable = Bool(False).tag(sync=True)
+
+    row2_title = Unicode("").tag(sync=True)
+    row2_text = Unicode("").tag(sync=True)
+    row2_unreliable = Bool(False).tag(sync=True)
+
+    row3_title = Unicode("").tag(sync=True)
+    row3_text = Unicode("").tag(sync=True)
+    row3_unreliable = Bool(False).tag(sync=True)
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self._marks = {}
+        self._x, self._y = None, None  # latest known cursor positions
+
+        # subscribe/unsubscribe to mouse events across all existing viewers
+        for viewer in self.app._viewer_store.values():
+            self._create_viewer_callbacks(viewer)
+
+        # subscribe to mouse events on any new viewers
+        self.hub.subscribe(self, ViewerAddedMessage, handler=self._on_viewer_added)
+
+    def _create_viewer_callbacks(self, viewer):
+        if isinstance(viewer,
+                      (SpecvizProfileView,
+                       ImvizImageView,
+                       CubevizImageView,
+                       MosvizImageView,
+                       MosvizProfile2DView)):
```

Comment:
One more thought about future-proofing this for a time-series config (or other user-created extension). It would probably be better to have this list of viewer types stored as a variable that can be added to dynamically on loading a custom config, rather than hard-coded here.

Reply:
I really like this idea, but I actually think the hardcoded implementations and other instance checks won't work right now for a custom config to hook into this plugin if the user/developer overwrote this list.  We should eventually either write a generic fallback, or provide some way for custom configs to hook into this existing plugin without having to rewrite large chunks of the plugin in an inherited version, for example).

Reply:
not this specifically, but as we scope work for building general support for custom configs, we will need to address this (across all general plugins).